packages/suite/src/actions/suite/metadataLabelingActions.ts

```diff
@@ -529,6 +568,297 @@ export const addMetadata =
+ */
+const getMetadataFiles =
+    () =>
+    async (dispatch: Dispatch, getState: GetState): Promise<[string[], string[], string[]]> => {
+        const providerInstance = dispatch(
+            metadataProviderActions.getProviderInstance({
+                clientId: selectSelectedProviderForLabels(getState())!.clientId,
+                dataType: 'labels',
+            }),
+        );
+
+        if (!providerInstance) {
+            throw new Error('no provider instance');
+        }
+
+        // fetch list of all files saved withing currently selected provider for labeling
+        const files = await providerInstance.getFilesList().then(response => {
+            if (!response.success) {
+                dispatch(
+                    metadataProviderActions.handleProviderError({
+                        error: response,
+                        action: ProviderErrorAction.LOAD,
+                        clientId: providerInstance.clientId,
+                    }),
+                );
+
+                return;
+            }
+
+            // todo: imho [] should be default return, it should not be also nullable
+            return response?.payload || [];
+        });
+
+        // no files, fresh account, no metadata encryption version migration needed
+        if (!files?.length) {
+            return [[], [], []];
+        }
+
+        // todo: this is not future proof in case there is another encryption version
+        const [currentEncryptionFiles, restFiles] = arrayPartition(files, file =>
+            file.endsWith(`_v2.mtdt`),
+        );
+
+        const [renamedOldEncryptionFiles, oldEncryptionFiles] = arrayPartition(restFiles, file =>
+            file.endsWith(`_v1.mtdt`),
+        );
+
+        return [currentEncryptionFiles, oldEncryptionFiles, renamedOldEncryptionFiles];
+    };
+
+const createMigrationPromise =
+    (
+        entity: LabelableEntity,
+        prevEncryptionVersion: MetadataEncryptionVersion,
+        fetchData: boolean,
+        device: TrezorDevice,
+    ) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        const encryptionVersion = selectEncryptionVersion(getState());
+
+        if (!device?.state || !device.metadata[encryptionVersion]) {
+            return { success: false, error: 'device unexpected state' };
+        }
+
+        let selectedProvider = selectSelectedProviderForLabels(getState());
+
+        const prevData =
+            fetchData &&
+            (await dispatch(
+                fetchMetadata({
+                    entity,
+                    encryptionVersion: prevEncryptionVersion,
+                    provider: selectedProvider!,
+                }),
+            ));
+
+        const nextKeys = entity[encryptionVersion];
+
+        if (!nextKeys) {
+            return { success: false, error: 'next keys are missing' };
+        }
+
+        const dummy = { dummy: getWeakRandomId(getRandomNumberInRange(1, 100)) };
```

Comment:
Is there any reason to not-use `import { randomBytes } from 'crypto';` I think we shall have proper entropy here.

Reply:
Also, we should somehow make the `1-100` better. This may be very predictable. I think the dummy data shall be in similar scale as real data so it is not distinguishable.